### DISCUSSION TITLE
Customization best practices for simple cases

### DISCUSSION THREAD

**Discussion by dgorshkov** [inventree/InvenTree#3910] [Q&A] [answered]
Hi there,

We have a customer with fairly simple processes, and InvenTree would be an overkill for them as-is.
To illustrate, a few things that are not needed:
- everything around manufacturing: builds, BOMs etc.
- multiple locations
- multiple shipments per sales order.

When it comes to extending InvenTree, it seems quite straightforward, existing classes can be augmented, new django apps can be added, new plugins etc.

But what should be the approach when simplifying? We've tried removing some things, and ran into broken UI really quickly.

**Comment by eeintech** [comment, marked as answer]
@dgorshkov I would first look into permissions: https://docs.inventree.org/en/latest/settings/permissions/

The way they were implemented in InvenTree, they enabled a high-level granularity to the UI and access to certain features.
It is possible to go to lower levels of permissions in Django and really customize and fine-control the UI hiding specific functions and elements using this method, however it is quite challenging to maintain on your own and ensure good compatibility for versions to come.

**Reply by dgorshkov** [reply]
Permissions is a really cool angle, thanks a lot!

**Reply by eeintech** [reply]
Glad you find it useful, would love to know what you come up with. A light version of InvenTree might be useful to a lot of people :wink:

**Comment by dgorshkov** [comment]
A quick update:
After carefully considering our options, we decided to build a custom backend with the exact features we need and with GraphQL as the default API, instead of trying to cut off parts of Inventree and using its REST API. The frontend will be covered by AppSmith.
